packages/ppx/test/native/At_rule_test.re

```diff
@@ -5,87 +5,200 @@ let loc = Location.none;
+          [||],
+        ),
+      |])
+    ],
+  ),
+  (
+    "(all)",
+    [%expr [%cx "@media (all) {}"]],
+    [%expr CssJs.style([|CssJs.media({js|(all) |js}, [||])|])],
+  ),
+  (
+    "all and (not(hover))",
+    [%expr [%cx "@media all and (not(hover)){}"]],
+    [%expr CssJs.style([|CssJs.media({js|all and (not(hover))|js}, [||])|])],
+  ),
   (
     "not all and (monochrome)",
     [%expr [%cx "@media not all and (monochrome) {}"]],
     [%expr
-      CssJs.style([|CssJs.media({js|not all and (monochrome)|js}, [||])|])
+      CssJs.style([|CssJs.media({js|not all and (monochrome) |js}, [||])|])
     ],
   ),
   (
-    "print and (color)",
-    [%expr [%cx "@media print and (color) {}"]],
-    [%expr CssJs.style([|CssJs.media({js|print and (color)|js}, [||])|])],
+    "screen, print and (color)",
+    [%expr [%cx "@media screen, print and (color) {}"]],
+    [%expr
+      CssJs.style([|CssJs.media({js|screen, print and (color) |js}, [||])|])
+    ],
   ),
   (
-    "(max-height: $(wat)",
-    [%expr [%cx "@media (max-height: $(wat)) {}"]],
+    "(min-height: 680px), screen and (orientation: portrait)",
+    [%expr
+      [%cx "@media (min-height: 680px), screen and (orientation: portrait) {}"]
+    ],
     [%expr
       CssJs.style([|
-        CssJs.media({js|(max-height: |js} ++ wat ++ {js|)|js}, [||]),
+        CssJs.media(
+          {js|(min-height: 680px), screen and (orientation: portrait) |js},
+          [||],
+        ),
       |])
     ],
   ),
   (
-    "($(wat))",
-    [%expr [%cx "@media ($(wat)) {}"]],
+    "(not (color)) or (hover)",
+    [%expr [%cx "@media (not (color)) or (hover) {}"]],
     [%expr
-      CssJs.style([|CssJs.media({js|(|js} ++ wat ++ {js|)|js}, [||])|])
+      CssJs.style([|CssJs.media({js|(not (color)) or (hover) |js}, [||])|])
     ],
   ),
   (
-    "$(wat)",
-    [%expr [%cx "@media $(wat) {}"]],
-    [%expr CssJs.style([|CssJs.media(wat, [||])|])],
-  ),
-  (
-    "(max-height: $(wat))",
+    "(max-height: $(wat)",
     [%expr [%cx "@media (max-height: $(wat)) {}"]],
     [%expr
       CssJs.style([|
-        CssJs.media({js|(max-height: |js} ++ wat ++ {js|)|js}, [||]),
+        CssJs.media({js|(max-height: |js} ++ wat ++ {js|) |js}, [||]),
       |])
     ],
   ),
-  /* Nested is not supported in code-gen
-     https://www.w3.org/TR/mediaqueries-5/#media-conditions */
-  /* (
-       "(not (color)) and (not (hover))",
-       [%expr [%cx "@media (not (color)) and (not (hover)) {}"]],
-       [%expr CssJs.style( [|CssJs.media( {js|((not (color)) and (not (hover))|js}, [||])|])]
-     ), */
-  /* "@media screen, print and (color) {}" */
+  (
+    "$(wat)",
+    [%expr [%cx "@media $(wat) {}"]],
+    [%expr CssJs.style([|CssJs.media(wat, [||])|])],
+  ),
+  // (
+  //   "($(wat))",
+  //   [%expr [%cx "@media ($(wat)) {}"]],
+  //   [%expr
+  //     CssJs.style([|CssJs.media({js|(|js} ++ wat ++ {js|)|js}, [||])|])
+  //   ],
+  // ),
 ];
```

Comment:
We need tests for broken media queries such as:

```
@media (min-width: calc(50vw + 10px)); /* functions aren't valid values */
@media not((color)); /* whitespace must follow `not` */
@media (768px < = width < 1200px); /* cannot have a space between `<` and `=` */
```

Reply:
> @media (min-width: calc(50vw + 10px)); /* functions aren't valid values */

Why should functions be invalid values?

> @media not((color)); /* whitespace must follow `not` */

This has now been fixed locally, so I expect it to error when written

> @media (768px < = width < 1200px); /* cannot have a space between `<` and `=` */

This currently errors when written

Reply:
functions aren't part of the media-feature, defined by the spec: https://www.w3.org/TR/mediaqueries-5/#mq-features

- Can you add a case where there's no space after not?
- Can you add a case with broken `< =`?

Reply:
> @media (min-width: calc(50vw + 10px)); /* functions aren't valid values */

That shall be added later because `mf-value` is not parsed safely, so it accepts functions.

Reply:
Can u add a comment on the parser?

Reply:
There is a comment on line 166 in the css_parser file